smartsim/settings/containers.py

```diff
@@ -161,7 +161,7 @@ def _container_cmds(self, default_working_directory: str = "") -> t.List[str]:
             )
 
         # Construct containerized launch command
-        cmd_list = [singularity, "exec"]
+        cmd_list = [singularity if singularity else "", "exec"]
```

Comment:
`which("singularity")` might return a None, which bubbles out and mangles my nice `t.List[str]` that is ultimately passed back and into popen. 

I figured that without the "singularity" up front, the command was going to fail and I could avoid passing back a list that allows nulls. 

In the end, a None also causes the Popen to break. If we do want that to error out, shouldn't we raise an error here when `which("singularity")` is None instead of letting it get back to the popen?

Reply:
Per offline convo, fixing by defaulting to `singularity or "singularity"` for head/node which failures. Fantastic catch!